#!/usr/bin/env python
PACKAGE = "image_cloud"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("threshold1",    double_t,   0, "Canny edge filter threshold1", 50,  1, 400)
gen.add("threshold2",    double_t,	  0, "Canny edge filter threshold1", 200,  1, 800)
gen.add("kernel_size",   int_t,	  0, "Canny and Laplace kernel size", 5,  3, 50)
gen.add("L2gradient",	bool_t,	0,	"Canny use L2gradient", False)
gen.add("subscribe_topic",   str_t,   0, "Subscribe Topic",  "")
gen.add("publish_topic",     str_t,   0, "Publish Topic",  "")
gen.add("publish_color",    bool_t,   0, "publish as color image",  True)


filter_enum = gen.enum([ gen.const("Canny",    int_t, 0, "Canny"),
                       gen.const("Laplace",     int_t, 1, "Laplace")],
                     "select filter")

gen.add("filter", int_t, 0, "filter selector", 0, 0, 1, edit_method=filter_enum)

exit(gen.generate(PACKAGE, "image_clouud", "edge_detector"))
